gapii/cc/vulkan_mid_execution.cpp

```diff
@@ -235,39 +235,114 @@ void VulkanSpy::EnumerateVulkanResources(CallObserver* observer) {
                 &swapchain.second->mVulkanHandle);
         }
     }
-    for (auto& memory: DeviceMemories) {
-        auto& dm = memory.second;
-        VkDeviceMemory mem = memory.second->mVulkanHandle;
-        VkMemoryAllocateInfo allocate_info {
-            VkStructureType::VK_STRUCTURE_TYPE_MEMORY_ALLOCATE_INFO,
-            nullptr, // pNext TODO: Handle
-                    //     VK_STRUCTURE_TYPE_DEDICATED_ALLOCATION_MEMORY_ALLOCATE_INFO_NV
-            dm->mAllocationSize,
-            dm->mMemoryTypeIndex
+    // Recreate CreateBuffers
+    {
+      VkBufferCreateInfo buffer_create_info{};
+      buffer_create_info.msType =
+          VkStructureType::VK_STRUCTURE_TYPE_BUFFER_CREATE_INFO;
+      for (auto& buffer : Buffers) {
+        BufferInfo& info = buffer.second->mInfo;
+        std::vector<uint32_t> queues(info.mQueueFamilyIndices.size());
+        for (size_t i = 0; i < info.mQueueFamilyIndices.size(); ++i) {
+          queues[i] = info.mQueueFamilyIndices[i];
+        }
+        buffer_create_info.mflags = info.mCreateFlags;
+        buffer_create_info.msize = info.mSize;
+        buffer_create_info.musage = info.mUsage;
+        buffer_create_info.msharingMode = info.mSharingMode;
+        buffer_create_info.mqueueFamilyIndexCount = queues.size();
+        buffer_create_info.mpQueueFamilyIndices = queues.data();
+
+        VkDedicatedAllocationBufferCreateInfoNV dedicated_allocation_create_info{
+          VkStructureType::VK_STRUCTURE_TYPE_DEDICATED_ALLOCATION_BUFFER_CREATE_INFO_NV,  // sType
+            nullptr,         // pNext
+            VkBool32(true),  // dedicatedAllocation
```

Comment:
This may have been false the first time. We should probably mimic the original code as much as possible, so keep track of IF they used the struct, and then if they did, figure out if they actually enabled dedicated allocation.